lib/jekyll/hooks.rb

```diff
@@ -0,0 +1,103 @@
+module Jekyll
+  module Hooks
+    # Helps look up hooks from the registry by owner's class
+    OWNER_MAP = {
+      Jekyll::Site => :site,
+      Jekyll::Page => :page,
+      Jekyll::Post => :post,
+      Jekyll::Document => :document,
+    }.freeze
```

Comment:
What if I want to make a hook for my custom plugin? Like post_sass or something?